cast/src/main.rs

```diff
@@ -53,8 +54,14 @@ enum Commands {
 #[tokio::main]
 async fn main() -> Result<()> {
     let cli = Cli::parse();
+    let accounts_file_path = match cli.accounts_file_path {
+        Some(path) => path,
+        None => Utf8PathBuf::from(format!("{}/.starknet_accounts/starknet_open_zeppelin_accounts.json", dirs::home_dir().expect("failed to read the home directory").to_str().expect("failed to convert home directory to string"))),
```

Comment:
Why do we use this file if no account file is provided? I don't think we should expect this file in users home dir at all. We should match it later on if and only if the command being executes needs this argument to succeed -in this case the command should fail if the argument is not found.

Reply:
this is a sane default. if there's no flag and this file does not exist, we should fail. but I'd leave this file as default, that's just good ux and nothing out of the blue in my opinion

Reply:
I don't really think it is a good UX. 

First of all, the file we are expecting here is created when using `cairo-lang` afaik (but I may be wrong), which will be deprecated. 

Secondly, we are expecting a file in a home directory which is a little bit weird imo. 

Moreover, if this file does not exist a user will get an error message informing about the missing file. It does not inform about the actual problem - not providing the required argument. 

Last but not least, a user could just forget to provide 

If we want to have a default argument for the account file, I think we should:
a) include it in in the struct we use for args parsing - this way it is readable and won't be lost/forgotten inside the codebase + it will be displayed in the help message for the cli
b) document it in docs - I guess that will be done automatically using `clap`, so point a should satisfy it

Reply:
> First of all, the file we are expecting here is created when using cairo-lang afaik (but I may be wrong), which will be deprecated.

the file is for openzeppelin accounts, we will also provide a command to create such account in the future

> Secondly, we are expecting a file in a home directory which is a little bit weird imo.

this is the default path where it's being created, it's not up to us

> Moreover, if this file does not exist a user will get an error message informing about the missing file. It does not inform about the actual problem - not providing the required argument.

Here I agree. 

> Last but not least, a user could just forget to provide this arg and create an account they did not intend to create by accident.

I don't think it's possible.

> If we want to have a default argument for the account file, I think we should:
a) include it in in the struct we use for args parsing - this way it is readable and won't be lost/forgotten inside the codebase + it will be displayed in the help message for the cli,
b) document it in docs - I guess that will be done automatically using clap, so point a should satisfy it.

this PR changes exactly that behavior. I don't have strong feelings about this, but if you believe it should not be moved from args parsing, let's revert to that.

> Imo we should remove it entirely though.

strongly disagree

Reply:
> this PR changes exactly that behavior. I don't have strong feelings about this, but if you believe it should not be moved from args parsing, let's revert to that.

I insist that we do that - otherwise we will just forget about it + the automatic cli help generated by clap will not include that info.

Reply:
Yes, so the problem was `~` which was not read correctly, we need something like `/Users/root/home`, where `root` is your user name and it varies so this cannot be hard coded. Therefore, we need to obtain the home directory in the runtime and this could not be done in `#[clap]`'s `default_value`, which probably needs to be specified in the compile time (but I may be wrong).

Reply:
That's weird, it was working for me and Marcin. let's try to debug offline

Reply:
discussed offline, we're about to address this separately. [revert change with default path](https://github.com/software-mansion/protostar/pull/2138/commits/1239607d6eeedf59787c186728beec19d5d1a262)